IBX-3582: Extend catalog-related PHP API doc (#1784)
ezsystems/developer-documentation: docs/pim/catalog_api.md

---
description: Use PHP API to get information about and manage product catalogs.
---

# Catalog API

To get information about product catalogs and manage them, use `CatalogServiceInterface`.

## Get catalog

To get a single catalog, you can use `CatalogServiceInterface::getCatalog()` and provide it with catalog ID, or `CatalogServiceInterface::getCatalogByIdentifier()` and pass the identifier:

``` php
[[= include_file('code_samples/api/product_catalog/src/Command/CatalogCommand.php', 70, 72) =]]
```

## Get products in catalog

> To get products from a catalog, request the product query from the catalog object  with `catalog::getQuery()`.

Reviewer: Did you mean `\Ibexa\Contracts\ProductCatalog\Values\CatalogInterface::getQuery` ?

---
description: Use PHP API to get information about and manage product catalogs.
---

# Catalog API

To get information about product catalogs and manage them, use `CatalogServiceInterface`.

## Get catalog

> To get a single catalog, you can use `CatalogServiceInterface::getCatalog()` and provide it with catalog ID, or `CatalogServiceInterface::getCatalogByIdentifier()` and pass the identifier:

Reviewer 2: ```suggestion
To get a single catalog, use `CatalogServiceInterface::getCatalog()` and provide it with catalog ID, or `CatalogServiceInterface::getCatalogByIdentifier()` and pass the identifier:
```

``` php
[[= include_file('code_samples/api/product_catalog/src/Command/CatalogCommand.php', 76, 78) =]]
```

## Get products in catalog

To get products from a catalog, request the product query from the catalog object
with `Ibexa\Contracts\ProductCatalog\Values\CatalogInterface::getQuery()`.
Then, create a new `ProductQuery` based on it and run a product search with `ProductServiceInterface::findProduct()`:

``` php
[[= include_file('code_samples/api/product_catalog/src/Command/CatalogCommand.php', 80, 86) =]]
```

## Create catalog

> To create a catalog, you need to prepare a `CatalogCreateStruct`, with the identifier, name and description and Criteria for filtering products.

Reviewer 2: ```suggestion
To create a catalog, you need to prepare a `CatalogCreateStruct` that contains: identifier, name, description and Criteria for filtering products.
```

Then, pass this struct to `CatalogServiceInterface::createCatalog()`:

``` php
[[= include_file('code_samples/api/product_catalog/src/Command/CatalogCommand.php', 66, 74) =]]
```

## Update catalog

> You can update an existing catalog by using `CatalogServiceInterface::updateCatalog()`,
> to which you must pass the catalog object and a `CatalogUpdateStruct`.

Reviewer 2: ```suggestion
Use `CatalogServiceInterface::updateCatalog()` to update an existing catalog.
You must pass the catalog object and a `CatalogUpdateStruct` to the interface.
```

> You can update an existing catalog by using `CatalogServiceInterface::updateCatalog()`,
> to which you must pass the catalog object and a `CatalogUpdateStruct`.

Author: ```suggestion
Use `CatalogServiceInterface::updateCatalog()` to update an existing catalog.
You must pass the catalog object and a `CatalogUpdateStruct` to the method.
```
